File: Java/lab14/src/main/java/org/example/Notification.java
package org.example;

public abstract class Notification {
    private final int prority;
    private Notification next;

    public Notification(int prority) {
        this.prority = prority;
    }

    public void setNext(Notification next) {
        this.next = next;
    }
    public void giveToNextRaopirman(String message, int level){
        if(level>=prority){
            printMessage(message);
        }

        if(next!=null){
            next.giveToNextRaopirman(message,level);
        }
    }
    public abstract void printMessage(String message);
}

File: Python/lab14/Notification.py
from abc import ABC, abstractmethod

class Notification(ABC):
    _next = None

    def __init__(self, priority: int):
        self.priority = priority

    def set_next(self, next):
        self._next = next

    def give_to_next_repairman(self,message:str,level:int):
        if level>=self.priority:
            self.print_message(message)
        if self._next!=None:
            self._next.give_to_next_repairman(message,level)

    @abstractmethod
    def print_message(self,message:str):
        pass